### PR TITLE
feat(nav): customizable bottom nav primary slots on phone

### PR DESCRIPTION
**Pull request by ericgriffin**
## Summary

- Phone users can now customize which 3 destinations occupy slots 2-4 of the bottom `NavigationBar`. Home (slot 1) and More (slot 5) stay pinned. Wide-screen `NavigationRail` is unaffected.
- Preference is global, stored as JSON in the Drift `settings` key-value table via `AppSettingsRepository` (same pattern as `shareByDefault`).
- Customization UI lives at Settings → Appearance → Navigation bar: a `ReorderableListView` with drag-and-drop and accessible move-up / move-down buttons, plus "Reset to defaults".

**Design spec:** `docs/superpowers/specs/2026-04-20-bottom-nav-customization-design.md`
**Implementation plan:** `docs/superpowers/plans/2026-04-20-bottom-nav-customization.md`

## What changed

| Layer | Files |
|---|---|
| Registry | `lib/shared/widgets/nav/nav_destinations.dart` — 14-entry `NavDestination` list (13 routable + `more` sentinel); pure `normalizeNavPrimaryIds` function |
| Data | `lib/features/settings/data/repositories/app_settings_repository.dart` — `getNavPrimaryIdsRaw` / `setNavPrimaryIds` |
| State | `lib/shared/widgets/nav/nav_primary_provider.dart` — `NavPrimaryIdsNotifier` + derived providers |
| UI | `lib/features/settings/presentation/pages/nav_customization_page.dart` (new); `lib/features/settings/presentation/pages/appearance_page.dart` (entry tile); `lib/shared/widgets/main_scaffold.dart` (mobile branch rewire) |
| Router | `lib/core/router/app_router.dart` — `/settings/appearance/navigation` |
| l10n | 9 new keys across 11 ARB files + regenerated bindings |

## Test plan

- [x] Registry + normalizer unit tests: 16 cases (uniqueness, kebab-case, canonical order, 8 normalizer edge cases)
- [x] Repository round-trip tests: 5 cases including invalid-JSON recovery
- [x] Notifier + derived provider tests: 8 cases including async load normalization
- [x] `NavCustomizationPage` widget tests: 11 cases (pinned rows, divider, reset enable/disable, drag logic, a11y buttons)
- [x] `MainScaffold` mobile-nav tests: 9 cases (default vs custom primary, wide-screen unaffected, tap-navigates, overflow sheet dynamic)
- [x] Integration test: 2 cases (customize → verify; reset → verify)
- [x] Full suite: 7098 tests pass
- [x] `flutter analyze`: clean
- [x] `dart format`: clean
- [x] Manual: reorder on phone, resize to wide (rail unchanged), resize back (custom order persists)

## Follow-ups (not in this PR)

- Locale-independent finders in `integration_test/nav_customization_test.dart` (currently use `find.byTooltip('Back')` and English text labels; works today because all locales ship English fallbacks)
- Consolidate wide-screen rail onto `kNavDestinations` to eliminate the hardcoded 13-item switch in `_calculateSelectedIndex`
- Spec typo: "Movable set: 11 entries" should be 12

### CODE REVIEW

File: lib/shared/widgets/nav/nav_destinations.dart [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+  /// When `true`, this destination cannot be moved between primary and overflow.
+  final bool isPinned;
+}
+
+/// The complete, ordered list of nav destinations in default wide-screen order.
+///
+/// Length is **14** — 13 routable destinations plus the `more` sentinel.
+final List<NavDestination> kNavDestinations = [
+  NavDestination(
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/shared/widgets/nav/nav_destinations.dart:43]: Fixed in fde678a7a29 — `kNavDestinations` now wraps its list literal in `List.unmodifiable(...)` matching the `app_theme_registry.dart` pattern.

File: lib/shared/widgets/nav/nav_destinations.dart [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    label: (l10n) => l10n.nav_more,
+    isPinned: true,
+  ),
+];
+
+/// The 12 ids that can be moved between primary slots and overflow.
+final List<String> movableNavIds = kNavDestinations
+    .where((d) => !d.isPinned)
+    .map((d) => d.id)
+    .toList(growable: false);
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/shared/widgets/nav/nav_destinations.dart]: Fixed in fde678a7a29 — replaced `.toList(growable: false)` with `List.unmodifiable(...)`, which also blocks element assignment.

File: lib/shared/widgets/nav/nav_destinations.dart [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+/// [defaults] must contain at least 3 ids from [movableIds]; otherwise this
+/// throws [ArgumentError]. Callers should pass [kDefaultPrimaryIds].
+List<String> normalizeNavPrimaryIds({
+  required List<String> stored,
+  required List<String> movableIds,
+  required List<String> defaults,
+}) {
+  assert(defaults.length >= 3, 'defaults must contain at least 3 ids');
+  for (final id in defaults.take(3)) {
+    if (!movableIds.contains(id)) {
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/shared/widgets/nav/nav_destinations.dart]: Fixed in fde678a7a29 — replaced the assert with an `ArgumentError.value` throw that runs in release too. The membership check on the first 3 defaults was already a runtime throw; this closes the length-precondition gap.

File: docs/superpowers/specs/2026-04-20-bottom-nav-customization-design.md [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+
+- Customizing the wide-screen `NavigationRail` (phone-only scope).
+- Reordering items inside the "More" sheet (overflow stays in canonical order).
+- Per-diver customization (preference is global).
+- Changing which screens exist; only their placement in the nav.
+
+## User flow
+
+1. User opens **Settings → Appearance → Navigation bar**.
+2. The page shows a single list: a pinned Home row at the top, 11 movable destinations in the middle (via `ReorderableListView`), a pinned More row at the bottom.
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply docs/superpowers/specs/2026-04-20-bottom-nav-customization-design.md]: Fixed in fde678a7a29 — 11 → 12. Also fixed the same typo at line 77 and a stale `itemCount` example (12 → 13 because 12 movable + 1 divider).

File: docs/superpowers/specs/2026-04-20-bottom-nav-customization-design.md [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+  });
+}
+```
+
+The constructor is `const`; the top-level list `kNavDestinations` is `final` (closures prevent the list from being `const`).
+
+Ids are lowercase, kebab-cased (matching the route slugs). The 13 routable ids are:
+`dashboard`, `dives`, `sites`, `trips`, `equipment`, `buddies`, `dive-centers`, `certifications`, `courses`, `statistics`, `planning`, `transfer`, `settings`.
+
+A 14th entry — the `more` sentinel — is included in `kNavDestinations` with `route: ''` and `isPinned: true`; it represents the overflow control, not a destination. Total registry size: **14 entries** (13 routable + 1 sentinel). Movable set: **11 entries** (all except `dashboard` and `more`).
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply docs/superpowers/specs/2026-04-20-bottom-nav-customization-design.md]: Fixed in fde678a7a29 — 11 → 12.

File: lib/features/settings/presentation/pages/nav_customization_page.dart [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+  /// through to the notifier, and rolls back with a SnackBar on failure.
+  Future<void> _commitReorder(int oldIndex, int newIndex) async {
+    final previous = _local!;
+    final newList = applyReorderPreservingDivider(
+      movable: previous,
+      dividerIndex: _dividerIndex,
+      oldIndex: oldIndex,
+      newIndex: newIndex,
+    );
+    if (identical(newList, previous)) return; // no-op reorder
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/settings/presentation/pages/nav_customization_page.dart]: Fixed in 782a586d91a — replaced `identical(newList, previous)` with `listEquals(newList, previous)`. You're right that `applyReorderPreservingDivider` returns a fresh list for any non-divider drag (it always goes through `List<String>.from(movable)` + removeAt + insert), so the old check only caught the divider-drag case. Added a unit test documenting that same-position drags (e.g. oldIndex=4, newIndex=4) produce a content-equal but non-identical list — so if the underlying function ever changed to reuse the input instance, tests would still enforce the right semantics.
